recipes/openblas/all/conanfile.py

```diff
@@ -19,134 +24,276 @@ class OpenblasConan(ConanFile):
-        return "source_subfolder"
+    def _openmp_runtime(self):
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            if self.settings.compiler == "gcc":
+                if Version(self.settings.compiler.version).major > "7":
+                    return "gomp"
+            if self.settings.compiler == "clang":
+                if Version(self.settings.compiler.version).major > "8":
+                    return "omp"
+        return None
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _fortran_compiler(self):
+        comp_exe = self.conf.get("tools.build:compiler_executables")
+        if comp_exe and 'fortran' in comp_exe:
+            return comp_exe["fortran"]
+        return environ.get('FC')
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+        # Build LAPACK by default if possible w/o Fortran compiler
+        if Version(self.version) >= "0.3.21":
+            self.options.build_lapack = True
+
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        if self.info.options.build_lapack:
+            # Capture fortran compiler in package id (if found)
+            f_compiler = self._fortran_compiler
+            if not f_compiler:
+                self.output.warning("None or unknown fortran compiler was used.")
+                f_compiler = True
+            self.info.options.build_lapack = f_compiler
 
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
+        if cross_building(self, skip_x64_x86=True):
             raise ConanInvalidConfiguration("Cross-building not implemented")
 
+        if self.options.use_thread and self.options.use_openmp:
+            raise ConanInvalidConfiguration("Both 'use_thread=True' and 'use_openmp=True' are not allowed")
+
+        if self.settings.os == "Windows" and self.options.build_lapack and self.options.use_openmp:
+            # In OpenBLAS cmake/system.cmake: Disable -fopenmp for LAPACK Fortran codes on Windows
+            self.output.warning("OpenMP is disabled on LAPACK targets on Windows")
+
+    def build_requirements(self):
+        if self.options.build_lapack and self.settings.os == "Windows":
+            self.tool_requires("ninja/1.11.1")
+
+    def system_requirements(self):
+        apt = package_manager.Apt(self)
+        dnf = package_manager.Dnf(self)
+        pacman = package_manager.PacMan(self)
+        # pkg = package_manager.Pkg(self) # TODO
+        yum = package_manager.Yum(self)
+        zypper = package_manager.Zypper(self)
+
+        if self.info.options.build_lapack == "gfortran":
```

Comment:
This isn't a valid `build_lapack` option in the current configuration

Reply:
That is true, but I have attempted to make use of `self.info`, for which is `options.build_lapack` is possibly extended in the [package_id](https://docs.conan.io/2/reference/conanfile/methods/package_id.html#package-id) function. It is a (sneaky) way to allow more binary package configurations without adding any option items. The main benefit is to avoid an exhaustive list of fortran compilers to the `build_lapack` option, but retain distinct binary packages depending on fortran compiler used (including runtime system packages). The correct way would be to extend `settings.yaml` with fortran compilers and distribute this, but this is not practical. The drawback is that a consumer would need to set either the environment variable `FC` or populating 'fortran' of `tools.build:compiler_executables` for this to be picked up correctly.

Reply:
I have removed system_requirements in the latest commit. If compiled with Fortran, the user is responsible for installing/deploying runtime dependencies. I agree that my suggested approach was incomplete and could cause troubles such as those you pointed out.

I have not yet added any new option such as `with_fortran` and have kept the `build_lapack` manipulation in package_id. I did not fully understand your suggestion regarding this. My intention with this is to "expose" different binary packages, at least to differentiate packages built with `build_lapack=True`,  but without Fortran (since 0.3.23) from those actually using Fortran.

Reply:
I guess the point I was making was that a `with_fortran` option might be more transparent as to what's causing a change in the package id. What you've got here is a clever solution but is a deviation from what I've normally seen in cci recipes so wasn't quite as intuitive. Given that this is to modify the package id based on compiler that hasn't been modelled though, I think that perhaps what you've got is a better solution than I proposed with `with_fortran`. I also hadn't considered that you'd already put in place a mechanism to propagate the system lib dependency downstream by reading from the `FORTRAN_COMPILER` file you create. Fine to disregard my suggestion here.